File: csharp/2007/aliens-vs-predator---requiem-add.cs
using System;
using System.Threading.Tasks;
using Amazon.DynamoDBv2;
using Amazon.DynamoDBv2.Model;
using System.Collections.Generic;

namespace MovieDatabase
{
    class Program
    {
        static async Task Main(string[] args)
        {
            // Create a MovieRepository instance
            var movies = new MovieRepository();

            // Add "Aliens vs Predator - Requiem" to the database
            // This demonstrates how to insert a new item into DynamoDB
            await movies.InsertAsync(
                title: "Aliens vs Predator - Requiem",
                year: 2007,
                plot: "Warring alien and predator races descend on a rural US town, where unsuspecting residents must band together for any chance of survival.",
                rating: 4.7
            );

            // Confirm that the movie was added by retrieving it
            var movie = await movies.SelectAsync(
                title: "Aliens vs Predator - Requiem",
                year: 2007
            );

            if (movie != null)
            {
                // The movie was found
                Console.WriteLine($"Movie found: {movie}");
            }
            else
            {
                // The movie was not found
                Console.WriteLine("Movie not found");
            }
        }
    }
}

File: python/2007/aliens-vs-predator---requiem-add.py

from movies import MovieRepository

# Create a new movie repository using the default configuration
movies = MovieRepository()

# Add Aliens vs Predator - Requiem to the database
movies.insert(
    title="Aliens vs Predator - Requiem", 
    year=2007, 
    plot="Warring alien and predator races descend on a rural US town, where unsuspecting residents must band together for any chance of survival.", 
    rating=4.7
)

# Confirm that the movie was added
movie = movies.select(
    title="Aliens vs Predator - Requiem", 
    year=2007
)
if movie:
    # The movie was found
    print(f"Movie found: {movie}")
else:
    # The movie was not found
    print("Movie not found")
    